spec/datadog/appsec/contrib/rails/integration_test_spec.rb

```diff
@@ -0,0 +1,284 @@
+# typed: ignore
+
+require 'datadog/tracing/contrib/rails/rails_helper'
+
+require 'datadog/appsec'
+
+RSpec.describe 'Rails integration tests' do
+  include Rack::Test::Methods
+
+  let(:sorted_spans) do
+    chain = lambda do |start|
+      loop.with_object([start]) do |_, o|
+        # root reached (default)
+        break o if o.last.parent_id == 0
+
+        parent = spans.find { |span| span.span_id == o.last.parent_id }
+
+        # root reached (distributed tracing)
+        break o if parent.nil?
+
+        o << parent
+      end
+    end
+    sort = ->(list) { list.sort_by { |e| chain.call(e).count } }
+    sort.call(spans)
+  end
+
+  let(:rack_span) { sorted_spans.reverse.find { |x| x.name == Datadog::Tracing::Contrib::Rack::Ext::SPAN_REQUEST } }
```

Comment:
Are there multiple rack spans? If so, I'm curious why, and if not, I believe this whole thing can be somewhat simplified :)

Reply:
Yes, on nested Rack apps, which I sadly have not added the case for (yet!).

Reply:
Uhh. That seems intuitively a bit weird... e.g. I guess it's an artifact of how we come in as a middleware, but as a user I'd still expect to have one single rack span even with nested apps.

(But I guess nothing to do for this PR)

Reply:
(we can discuss this further internally if you wish)

Reply:
You make a good point. I guess part of the trade-off here is that the "rack span" is pulling double duty: e.g. if I'm using "puma", there is no "puma span", only the "rack span".

My "intuitively a bit weird" comment was from that POV -- if I'm thinking of the "rack span" as the "puma span", then even if we do the weirdest routing things, there should only be one "puma span".

But when considering the "rack span" as the span representing a stack, which can have substacks, your point makes a lot of sense.

So... I think the current behavior seems reasonable enough, we have bigger fish to fry ;)